### DISCUSSION TITLE
Returns completions when read-file-name must-match returns nil

### DISCUSSION THREAD

**Discussion by bsvh** [minad/vertico#519] [Q&A]
I'm using `read-file-name` in a function where I want to restrict selections to certain file types. I accomplished this using the following code

```emacs-lisp
(defun my/test (f) 
  (when (and (not (file-directory-p f))
             (file-exists-p f)
             (string-match-p "\\.\\(?:org\\|txt\\)\\'" f))

    f))

(read-file-name "Document: "
                nil
                nil
                #'my/test
                nil
                (lambda (f) (or (file-directory-p f)
                                (my/test f))))
```

When using builtin minibuffer modes , this only shows completions for txt, org files, and directories, but pressing enter on a directories does nothing. With vertico, completions for txt, org files, and directories are properly shown, but pressing enter on a directory returns that directory. Is this intended?

**Comment by minad** [comment]
Thanks. Fixed in https://github.com/minad/vertico/commit/93fe123599695dd992a73f1e9daae06ea86172fe. Turns out the behavior of the REQUIRE-MATCH argument was changed due to bug
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=66187. Vertico replicated the old behavior.
